### DISCUSSION TITLE
Is there a way to get the htmlFieldPrefix in widget.edit

### DISCUSSION THREAD

**Discussion by SteeveDumais** [OrchardCMS/OrchardCore#15860] [Q&A]
I am trying to get the html prefix of a widget so that I can manually create the field in Widget-{myWidget}.edit.cshtml.

All my widgets will be in react and I need to name them correctly so that the ModelUpdate will pick up the modification in edit mode.

**Comment by ns8482e** [comment]
@SteeveDumais  For example, If your widget is called `News` create template with `Content-News.Edit.cshtml`

**Comment by SteeveDumais** [comment]
I have created the file Content-{MyWidget}.edit.cshtml but it is not being hit.  Is there no way to get the value on the widget.edit.cshtml page?

**Comment by ns8482e** [comment]
Hmm! So If your Content Type `News` has  Stereotype `Widget` then It will take `Widget-News.Edit.cshtml`. But if  stereotype  is blank then it will take `Content-News.Edit.cshtml`

`ShapeTable` is created at start of the application, so if you add new shapes in `Views`  folder, you may need to restart the application.

**Comment by SteeveDumais** [comment]
Where in the model is the prefixe when I go to Widget-News.Edit.cshtml.  I have modified the ContentCard-FlowPart.Edit.cshtml to force the Model.ContentEditor.Metadata.Prefix = Model.PrefixValue.  I was hoping that I did not have to do that step and that Model.ContentEditor had that value somewhere or it was present int he viewContext

**Comment by ns8482e** [comment]
@SteeveDumais I don't think that you need to change the prefix value. 
Can you explain what you are trying to achieve?

**Comment by SteeveDumais** [comment]
I am creating a contentCard that has a Content of stereotype widget.  When I display the widget it is a react component.  In order for the values to be saved the widget components need to be named "prefixValue. the field name", where the prefixValue is coming from the contentCard.  I was hoping there was a way to reference the contentCard prefix in the widget-myWidget.Edit.cshtml page

**Comment by ns8482e** [comment]
@SteeveDumais when you use `Widget-{contenttype}.Edit.cshtml`, the Model is an editor ZoneHolding shape of your content type . You can access fields and parts using `Model.Content` or `Model.Parts` shapes.

**Comment by ns8482e** [comment]
@SteeveDumais with recent change in PR #5989 , now, PrefixValue is ContentItemId for existing Item.
